WordPress/openverse-frontend
# Convert `media` store to an options store

## Fixes


Related to #1254 by @obulat
Fixes #1235 by @sarayourfriend 

## Description


This PR does two main things:
1. converts media store to an options store to fix the server state hydration issues.
2. Extracts the `media-item` state (for single image/audio pages).

The extraction of the `media-item` makes it possible to re-use the media item information from the search results on the single media detail page, instead of fetching it every time, even when we already have this data in the store.
It also refactors the single image detail page to get the media data in the `asyncData`.

## Testing Instructions

The media store is the backbone of the app state, so all of these critical paths could be affected, and should not have any problems after this PR:
Searching for audio/image from homepage, search page and content pages on server-rendered (opened URL directly) and client-rendered (opened by following an internal link) pages.
Server-rendered search pages (All Content, image and audio) with no results, with fewer than 20 results and with more than 20 results should be interactive (at the minimum, you should be able to click on the content switcher and change the search type.
Clicking on "Load more".
Selecting a filter and deselecting it, or going back using the browser back button.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
